packages/pages/src/dev/server/middleware/serverRenderModule.ts

```diff
@@ -0,0 +1,42 @@
+import { RequestHandler } from "express-serve-static-core";
+import { ViteDevServer } from "vite";
+import { ProjectStructure } from "../../../common/src/project/structure.js";
+import send404 from "./send404.js";
+import { getModuleFilepathsFromProjectStructure } from "../../../common/src/module/internal/getModuleFilepaths.js";
+import { findModuleInternal } from "../ssr/findMatchingModule.js";
+import serveModule from "./serveModule.js";
+
+type Props = {
+  vite: ViteDevServer;
+  projectStructure: ProjectStructure;
+};
+
+export const serverRenderModule =
+  ({ vite, projectStructure }: Props): RequestHandler =>
+  async (req, res, next): Promise<void> => {
+    try {
+      const moduleName = req.originalUrl
+        .substring(req.originalUrl.lastIndexOf("modules/") + 8)
```

Comment:
You can do lastIndexOf("/") + 1

Reply:
That's fine. We're not allowing multi-level deep urls like that. Maybe that will be a feature someday but not something to worry about now.